tests/extension/oneapi_root_group/root_group.cpp

```diff
@@ -0,0 +1,256 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Copyright (c) 2023 The Khronos Group Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+*******************************************************************************/
+
+#include "../../common/common.h"
+
+namespace root_group::tests {
+
+#ifdef SYCL_EXT_ONEAPI_ROOT_GROUP
+
+template <typename KernelName, size_t... Dims>
+static void check_root_group_api() {
+  constexpr int Dimensions = sizeof...(Dims);
+  auto q = sycl_cts::util::get_cts_object::queue();
+  auto bundle =
+      sycl::get_kernel_bundle<sycl::bundle_state::executable>(q.get_context());
+  auto kernel = bundle.get_kernel<KernelName>();
+  auto maxWGs = kernel.template ext_oneapi_get_info<
+      sycl::ext::oneapi::experimental::info::kernel_queue_specific::
+          max_num_work_group_sync>(q);
+  REQUIRE(maxWGs >= 1);
+  auto local_range = sycl::range<Dimensions>(Dims...);
+  auto global_range = local_range;
+  global_range[0] *= maxWGs;
+  REQUIRE(global_range.size() == local_range.size() * maxWGs);
+  auto nd_range = sycl::nd_range<Dimensions>{global_range, local_range};
+  auto props = sycl::ext::oneapi::experimental::properties{
+      sycl::ext::oneapi::experimental::use_root_sync};
+
+  struct checks {
+    enum {
+      get_group_id,
+      get_local_id,
+      get_group_range,
+      get_local_range,
+      get_max_local_range,
+      get_group_linear_id,
+      get_local_linear_id,
+      get_group_linear_range,
+      get_local_linear_range,
+      leader,
+      get_child_group___root_group,
+      get_child_group___group,
+      COUNT,
+    };
+  };
```

Comment:
Why not use a [scoped `enum`](https://en.cppreference.com/w/cpp/language/enum)?

```suggestion
  enum class checks {
    get_group_id,
    get_local_id,
    get_group_range,
    get_local_range,
    get_max_local_range,
    get_group_linear_id,
    get_local_linear_id,
    get_group_linear_range,
    get_local_linear_range,
    leader,
    get_child_group___root_group,
    get_child_group___group,
    COUNT,
  };
```

Reply:
Good point about `static_cast`, I forgot about it. In that case I agree that the current approach is better